boards/PDM/Src/Io/Io_Efuses.c

```diff
@@ -0,0 +1,227 @@
+#include "Io_Efuses.h"
+#include "configs/Io_EfuseConfig.h"
+
+#pragma GCC diagnostic ignored "-Wconversion"
+
+/**
+ * Set the current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  AUX1_CURRENT_SENSE_CHANNEL - Current sensing for Aux 1
+ *                  AUX2_CURRENT_SENSE_CHANNEL - Current sensing for Aux 2
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection);
+
+/**
+ * Exit fail-safe mode and disable the watchdog timer
+ * @return EXIT_CODE_OK if the efuse exited fail-safe mode
+ *         EXIT_CODE_TIMEOUT if the one of the SPI writes timed-out
+ *         EXIT_CODE_UNIMPLEMENTED if the efuse failed to exit fail-safe mode
+ */
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(void);
+
+/**
+ * Write data to a specific Serial Input register on the Aux1/Aux2 Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @return EXIT_CODE_OK if the write was successful
+ *         EXIT_CODE_TIMEOUT if the SPI write timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2WriteRegister(
+    uint8_t  register_address,
+    uint16_t register_value);
+
+/**
+ * Read data from a specific Serial Output register on the Aux1/Aux2 Efuse
+ * @param register_address Serial Output register being read from
+ * @param register_value register contents of Serial Output register
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ReadRegister(
+    uint8_t   register_address,
+    uint16_t *register_value);
+
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection)
+{
+    uint16_t register_value = 0x0000;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, &register_value));
```

Comment:
If the constants aren't specific to Aux1 Aux2 this can also be moved to Io_Efuse.c (Assuming that the read register function here can be moved to Io_Efuse.c)

Reply:
I see now, DONE